### DISCUSSION TITLE
Add distinctive color for the late and early tapped 100s and 50s.

### DISCUSSION THREAD

**Discussion by execeval** [ppy/osu#22874] [Ideas]
This change may appear somewhat confusing at first, but players can always disable it in settings or customize it with skins. The main idea behind it is to provide clearer feedback on tapping timing for the early and late hits of 100s and 50s.

While this may seem like a minor detail, it can actually be quite helpful for the brain to learn and improve its motor skills. By distinguishing between hitting too early or too late, players can better understand and correct their mistakes, which in turn can strengthen their muscle memory and overall performance.

Although UR bar can provide some indication of timing accuracy, it is not always easy to track in the heat of the moment, especially during intense maps. By adding distinctive colors for the early and late tapped 100s and 50s, players can receive more intuitive and immediate feedback on their timing, which can lead to faster and more accurate improvements.

Of course, not everyone may find this feature helpful or necessary, for this reason, player should be able to turn this feature off.

**Comment by monochrome22** [comment]
https://github.com/ppy/osu/discussions/22622
https://github.com/ppy/osu/discussions/22603
https://github.com/ppy/osu/discussions/14858
https://github.com/ppy/osu/discussions/14038
